Imagine/Filter/PostProcessor/CwebpPostProcessor.php

```diff
@@ -0,0 +1,230 @@
+     * automatically. The range is **0** (maximum preprocessing) to **100** (no preprocessing). The typical value is
+     * around **60**. Note that lossy with **-q 100** can at times yield better results.
+     *
+     * @var int
+     */
+    protected $nearLossless;
+
+    /**
+     * Specify the compression factor for RGB channels between **0** and **100**. The default is **75**.
+     *
+     * In case of lossy compression , a small factor produces a smaller file with lower quality. Best quality is
+     * achieved by using a value of **100**.
+     *
+     * In case of lossless compression (specified by the **-lossless** option), a small factor enables faster
+     * compression speed, but produces a larger file. Maximum compression is achieved by using a value of **100**.
+     *
+     *
+     * @var int
+     */
+    protected $q;
+
+    /**
+     * Specify the compression factor for alpha compression between **0** and **100**. Lossless compression of alpha is
+     * achieved using a value of **100**, while the lower values result in a lossy compression.
+     *
+     * @var int
+     */
+    protected $alphaQ;
+
+    /**
+     * Specify the compression method to use. This parameter controls the trade off between encoding speed and the
+     * compressed file size and quality. Possible values range from **0** to **6**. When higher values are used, the
+     * encoder will spend more time inspecting additional encoding possibilities and decide on the quality gain. Lower
+     * value can result in faster processing time at the expense of larger file size and lower compression quality.
+     *
+     * @var int
+     */
+    protected $m;
+
+    /**
+     * Specify the predictive filtering method for the alpha plane. One of **none**, **fast** or **best**, in
+     * increasing complexity and slowness order. Internally, alpha filtering is performed using four possible
+     * predictions (none, horizontal, vertical, gradient). The **best** mode will try each mode in turn and pick the
+     * one which gives the smaller size. The **fast** mode will just try to form an a priori guess without testing all
+     * modes.
+     *
+     * @var string
+     */
+    protected $alphaFilter;
+
+    /**
+     * Specify the algorithm used for alpha compression: **0** or **1**. Algorithm **0** denotes no compression, **1**
+     * uses WebP lossless format for compression.
+     *
+     * @var int
+     */
+    protected $alphaMethod;
+
+    /**
+     * Preserve RGB values in transparent area. The default is off, to help compressibility.
+     *
+     * @var bool
+     */
+    protected $exact;
+
+    /**
+     * A comma separated list of metadata to copy from the input to the output if present. Valid values: **all**,
+     * **none**, **exif**, **icc**, **xmp**.
+     *
+     * Note that each input format may not support all combinations.
+     *
+     * @var string
+     */
+    protected $metadata;
```

Comment:
i would prefer to define this as an array of string. the comma separated list is the way of telling it to the cwebp binary, but not a good representation of the configuration in php.

Reply:
Done!